From 3b70553a4e42474d91aec05050c02c72534a30fe Mon Sep 17 00:00:00 2001
From: William Horvath <william@horvath.blog>
Date: Fri, 7 Nov 2025 13:54:59 -0800
Subject: [PATCH] Use RIDEV_NOLEGACY for Windows raw mouse input

---
 src/video/windows/SDL_windowsrawinput.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/video/windows/SDL_windowsrawinput.c b/src/video/windows/SDL_windowsrawinput.c
index f1e72a220..199d1ca8b 100644
--- a/src/video/windows/SDL_windowsrawinput.c
+++ b/src/video/windows/SDL_windowsrawinput.c
@@ -97,7 +97,7 @@ static bool UpdateRawInputDeviceFlags(HWND window, Uint32 last_flags, Uint32 new
         devices[count].usUsage = USB_USAGE_GENERIC_MOUSE;
 
         if (new_flags & ENABLE_RAW_MOUSE_INPUT) {
-            devices[count].dwFlags = 0;
+            devices[count].dwFlags = RIDEV_NOLEGACY;
             devices[count].hwndTarget = window;
         } else {
             devices[count].dwFlags = RIDEV_REMOVE;
-- 
2.53.0

